sample_input_files/calculateSchoolAgeMixing.py

```diff
@@ -0,0 +1,246 @@
+"Perth & Kinross" : "Perth and Kinross"}
+
+# Get the number of schools in each local authority from Table 1
+num_school_worksheet = workbook["Table 1"]
+for row in num_school_worksheet.iter_rows(min_row=4, max_col=8, max_row=40):
+
+    if row[0].value is not None:
+
+        authority_name = row[0].value
+        if authority_name in authority_name_map:
+            authority_name = authority_name_map[row[0].value]
+
+        localAuthorities[authority_name].primary_schools = int(row[1].value)
+        localAuthorities[authority_name].secondary_schools = int(row[2].value)
+        localAuthorities[authority_name].total_schools = int(row[4].value)
+
+# Get the population of school pupils from Table 8
+school_pop_worksheet = workbook["Table 8"]
+for row in school_pop_worksheet.iter_rows(min_row=5, max_col=17, max_row=2496):
+
+    authority_name = row[2].value
+    if authority_name in authority_name_map:
+        authority_name = authority_name_map[row[2].value]
+
+    num_pupils = 0
+    try:
+        num_pupils = int(row[14].value)
+    except ValueError:
+        pass # there is a # in the Capacity percent column
+    if num_pupils > 0:
+        if "Primary" == row[3].value:
+            localAuthorities[authority_name].primary_school_total += num_pupils
+            localAuthorities[authority_name].primary_school_pupils.append(num_pupils)
+        elif "Secondary" == row[3].value:
+            localAuthorities[authority_name].secondary_school_total += num_pupils
+            localAuthorities[authority_name].secondary_school_pupils.append(num_pupils)
+        elif "Special" == row[3].value:
+            localAuthorities[authority_name].special_school_total += num_pupils
+            localAuthorities[authority_name].special_school_pupils.append(num_pupils)
+
+
+# The algorithm is now:
+# Assuming that an average class size is 30 pupils and that there is complete mixing within a class and 60% mixing
+# between classes in a school the mixing (ave number of contacts a pupil will have in a school) is
+# 30*1.0 + (school_size-30)*0.6
+# and multiplying this by the number of pupils in the school will give the number of contacts within the school.
+#
+# The number of contacts between schools in a local authority, assuming that the contact rate is 5%, is
+# 0.5 * 0.05 * (number of pupils in local authority - number pupils in the school)
+# where the factor of 0.5 ensures we do not double count between school contacts.
+#
+# The number of contacts is presented as the average number of contacts a school age child will have as a 
+# fraction (percentage) of the size of the school.
+#
+# The total number of contacts in a local authority is an average of all the contacts wєighted according to the 
+# school size.
+# 
+# A similar weighting across local authorities in a health board is done.
+
+# The weighting factors are asssumed to be:
+within_class_mixing = 1.0
+within_school_mixing = 0.4
+within_la_mixing = 0.01
+ave_class_size = 30
```

Comment:
Are these factors calculated from some source or are they just guesses? I think modelling just within class mixing is probably good enough. I think trying to add other types of mixing will just complicate the model without bringing more certainty.

Reply:
just guesses for now. The within_school and between school rates is really to have a weight when averaging over all schools in a local authority. If we do just within_class I would expect the rate to be ave_class_size/num_classes_in_the_school. You're right about over complicating the model (especially adding in rates that are just guesses) but just putting in within_class mixing will give a rate that is lower than reality.

Reply:
A similar consideration should be made when we calculate the mixing rates for adults (within_office?, within_factory? etc)